Framework/PythonInterface/plugins/algorithms/IntegratePeaksSkew.py

```diff
@@ -0,0 +1,714 @@
+                                 " integrated intensities.")
+
+    def validateInputs(self):
+        issues = dict()
+        # check peak size limits are consistent with window size
+        drows = self.getProperty("NRows").value
+        dcols = self.getProperty("NCols").value
+        nrow_max = self.getProperty("NRowMax").value
+        ncol_max = self.getProperty("NColMax").value
+        if nrow_max > 2 * drows + 1:
+            issues["NRowMax"] = "NRowMax exceeds window size."
+        if ncol_max > 2 * dcols + 1:
+            issues["NColMax"] = "NColMax exceeds window size."
+        npk_min = self.getProperty("NPixMin").value
+        if npk_min > (2 * drows + 1) * (2 * dcols + 1):
+            issues["NPixMin"] = "NPixMin exceeds number of pixels in the window."
+        return issues
+
+    def PyExec(self):
+        # get input
+        ws = self.getProperty("InputWorkspace").value
+        pk_ws = self.getProperty("PeaksWorkspace").value
+        # peak window parameters
+        frac_tof_window = self.getProperty('FractionalTOFWindow').value
+        dt0_over_t0 = self.getProperty("BackscatteringTOFResolution").value
+        dth = self.getProperty("ThetaWidth").value
+        drows = self.getProperty("NRows").value
+        dcols = self.getProperty("NCols").value
+        optimise_mask = self.getProperty("OptimiseMask").value
+        # peak mask validation
+        integrate_on_edge = self.getProperty("IntegrateIfOnEdge").value
+        nrows_edge = self.getProperty("NRowsEdge").value
+        ncols_edge = self.getProperty("NColsEdge").value
+        npk_min = self.getProperty("NPixMin").value
+        density_min = self.getProperty("DensityPixMin").value
+        nrow_max = self.getProperty("NRowMax").value
+        ncol_max = self.getProperty("NColMax").value
+        max_nvacancies = self.getProperty("NVacanciesMax").value
+        min_npixels_per_vacancy = self.getProperty("NPixPerVacancyMin").value
+        # peak finding
+        use_nearest = self.getProperty("UseNearestPeak").value
+        update_peak_pos = self.getProperty("UpdatePeakPosition").value
+        # plotting
+        plot_filename = self.getProperty("OutputFile").value
+
+        array_converter = InstrumentArrayConverter(ws)
+
+        # setup progress bar
+        prog_reporter = Progress(self, start=0.0, end=1.0, nreports=pk_ws.getNumberPeaks())
+
+        # Empty table workspace (clone and delete so as to preserve UB, sample, history etc.)
+        pk_ws_int = self.child_CloneWorkspace(InputWorkspace=pk_ws, OutputWorkspace="_temp")  # temp for ws in/out same
+        self.child_DeleteTableRows(TableWorkspace=pk_ws_int, Rows=range(pk_ws_int.getNumberPeaks()))
+        # get spectrum indices for all peaks in table
+        detids = pk_ws.column('DetID')
+        bank_names = pk_ws.column('BankName')
+        tofs = pk_ws.column('TOF')
+        if plot_filename:
```

Comment:
Perhaps this should also check that pk_ws has at least one peak so that an empty pdf is never created? One of my other comments is about adding some validation in the peak loop so perhaps this should only be created if the integration succeeds for at least one peak in pk_ws which would mean creating later on after the 1st peak is successfully integrated

Reply:
I have added a check that the peak workspace has at least one peak in the input validation - I still want to plot cases where the integration fails because the colorfill plot shows why it failed (i.e. if the mask wasn't valid), which may be informative to the user.

Reply:
I think if the detid check that you've added results in a "continue" on L640 then you could end up with an empty pdf? I agree that the extra validation closes off one route to getting an empty pdf so see how much time you have later this week for addressing this - can skip if run out of time